clusterloader2/pkg/measurement/common/prometheus_measurement.go

```diff
@@ -56,10 +59,21 @@ type prometheusMeasurement struct {
 }
 
 func (m *prometheusMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
-	if config.PrometheusFramework == nil {
+	prometheusClient, err := util.GetStringOrDefault(config.Params, "prometheusClient", "inCluster")
+	if err != nil {
+		return nil, err
+	}
+	if prometheusClient != "inCluster" && prometheusClient != "managed" {
+		return nil, fmt.Errorf("unknown Prometheus client")
+	}
+	if config.PrometheusFramework == nil && prometheusClient == "inCluster" {
 		klog.Warningf("%s: Prometheus is disabled, skipping the measurement!", config.Identifier)
 		return nil, nil
 	}
+	if cp := config.CloudProvider.Name(); cp != provider.GKEName && cp != provider.GCEName && prometheusClient == "managed" {
```

Comment:
Can we add flag to provider that specifies whether managed prometheus is supported and set it for gke and gce to true?

Reply:
Good suggestion, done.